api/controllers/models.py

```diff
@@ -415,3 +416,62 @@ def upload_to_s3(credentials):
             {"model_id": model.id, "s3_uri": f"s3://{bucket_name}/{s3_path}"}
         )
     )
+
+
+@bottle.get("/models/<mid:int>/deploy")
+@_auth.requires_auth
+def deploy_model_from_s3(credentials, mid):
+    # Authentication (only authenticated users can redeploy models for interaction)
+    u = UserModel()
+    user_id = credentials["id"]
+    user = u.get(user_id)
+    if not user:
+        logger.error("Invalid user detail for id (%s)" % (user_id))
+        bottle.abort(404, "User information not found")
+
+    m = ModelModel()
+    model = m.getUnpublishedModelByMid(mid)
```

Comment:
Do we want to be able to deploy published models too?

Reply:
I think we also deploy published models with this. I'm not sure why it is called `getUnpublishedModelByMid` since the code for it is 
`self.dbs.query(Model).filter(Model.id == id).one()` which seems like it just gets any model by mid, whether it is unpublished or not.